java-squid/src/main/java/org/sonar/plugins/java/api/JavaFileScannerContext.java

```diff
@@ -54,10 +54,26 @@
 
   File getFile();
 
-  List<Tree> getComplexity(Tree tree);
-
-  List<Tree> getMethodComplexity(ClassTree enclosingClass, MethodTree methodTree);
-
+  /**
+   * @deprecated As of release 3.6, replaced by {@link #getComplexityNodes(Tree)}
+   */
+  @Deprecated
+  int getComplexity(Tree tree);
+
+  /**
+   * @deprecated As of release 3.6, replaced by {@link #getMethodComplexityNodes(ClassTree, MethodTree)}
+   */
+  @Deprecated
+  int getMethodComplexity(ClassTree enclosingClass, MethodTree methodTree);
+
+  List<Tree> getComplexityNodes(Tree tree);
+
+  List<Tree> getMethodComplexityNodes(ClassTree enclosingClass, MethodTree methodTree);
+
+  /**
+   * @deprecated As of release 3.6, this method was not intended on a public interface
+   */
+  @Deprecated
   void addNoSonarLines(Set<Integer> lines);
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)